src/libs/actions/Session/index.js

```diff
@@ -464,38 +464,40 @@ const reauthenticatePusher = _.throttle(() => {
  * @param {Function} callback
  */
 function authenticatePusher(socketID, channelName, callback) {
-    Log.info('[PusherConnectionManager] Attempting to authorize Pusher', false, {channelName});
+    Log.info('[PusherAuthorizer] Attempting to authorize Pusher', false, {channelName});
 
     API.Push_Authenticate({
         socket_id: socketID,
         channel_name: channelName,
         shouldRetry: false,
         forceNetworkRequest: true,
     })
-        .then((data) => {
-            if (data.jsonCode === CONST.JSON_CODE.NOT_AUTHENTICATED) {
-                callback(new Error('Expensify session expired'), {auth: ''});
+        .then((response) => {
+            if (response.jsonCode === CONST.JSON_CODE.NOT_AUTHENTICATED) {
+                Log.hmmm('[PusherAuthorizer] Unable to authenticate Pusher because authToken is expired');
+                callback(new Error('Pusher failed to authenticate because authToken is expired'), {auth: ''});
 
                 // Attempt to refresh the authToken then reconnect to Pusher
                 reauthenticatePusher();
                 return;
             }
 
-            if (data.jsonCode !== CONST.JSON_CODE.SUCCESS) {
-                Log.hmmm('[PusherConnectionManager] Unable to authenticate Pusher for some reason other than expired session');
+            if (response.jsonCode !== CONST.JSON_CODE.SUCCESS) {
+                Log.hmmm('[PusherAuthorizer] Unable to authenticate Pusher for reason other than expired session');
+                callback(new Error(`Pusher failed to authenticate because code: ${response.jsonCode} message: ${response.message}`), {auth: ''});
                 return;
             }
 
             Log.info(
-                '[PusherConnectionManager] Pusher authenticated successfully',
+                '[PusherAuthorizer] Pusher authenticated successfully',
                 false,
                 {channelName},
             );
-            callback(null, data);
+            callback(null, response);
         })
         .catch((error) => {
-            Log.info('[PusherConnectionManager] Unhandled error: ', false, {channelName});
-            callback(error, {auth: ''});
+            Log.hmmm('[PusherAuthorizer] Unhandled error: ', {channelName, error});
+            callback(new Error('Push_Authenticate request failed'), {auth: ''});
```

Comment:
I'm not re-throwing this error because the error that can show up in a catch doesn't really offer much information about why the request failed anyway. It could have:

- run out of retries
- server returned 403 Forbidden header (which we can't tell from the JS)
    - User doesn't have access to a report
    - Channel name isn't valid
    - etc...